rust/src/passwd.rs

```diff
@@ -387,6 +389,40 @@ fn data_from_json(
         })
         .with_context(|| format!("failed to write /{}", &target_etc_filename))?;
 
+    // Regernate etc/{,g}shadow to sync with etc/{password,group}
+    let db = rootfs.open(target_etc_filename).map(BufReader::new)?;
+    let shadow_name = if target == "passwd" {
+        "shadow"
+    } else {
+        "gshadow"
+    };
+    let target_etc_shadow = format!("{}{}", dest_path, shadow_name);
+
+    match shadow_name {
+        "shadow" => {
+            let entries = nameservice::passwd::parse_passwd_content(db)?;
+            rootfs
+                .atomic_replace_with(&target_etc_shadow, |target_shadow| -> Result<()> {
+                    for user in entries {
+                        target_shadow.write_fmt(format_args!("{}:*::0:99999:7:::\n", user.name))?;
+                    }
+                    Ok(())
+                })
+                .with_context(|| format!("Writing {target_etc_shadow}"))?;
+        }
+        "gshadow" => {
+            let entries = nameservice::group::parse_group_content(db)?;
+            rootfs
+                .atomic_replace_with(&target_etc_shadow, |target_shadow| -> Result<()> {
+                    for group in entries {
+                        target_shadow.write_fmt(format_args!("{}:::\n", group.name))?;
+                    }
+                    Ok(())
+                })
+                .with_context(|| format!("Writing {target_etc_shadow}"))?;
+        }
+        _ => {}
```

Comment:
Same as above, this should never be reached and should fail.

Reply:
Yeah, `unreachable!` is probably OK here. 

(Or with more work, we could actually try to pass around an `enum PasswdKind { User, Group }`  instead?)

Reply:
Makes sense to me, updated, thanks!